catalyst/utils/config.py

```diff
@@ -257,6 +258,28 @@ def parse_config_args(*, config, args, unknown_args):
     return config, args
 
 
+def get_recursive_configs(config_path):
+    configs = []
+    config_ = safitty.load(config_path)
+    base_dir = os.path.dirname(config_path)
+
+    if "pre_configs" in config_:
+        pre_configs = config_["pre_configs"]
+        for pre_config in pre_configs:
+            pre_config = os.path.join(base_dir, pre_config)
+            configs += get_recursive_configs(pre_config)
+
+    configs += [config_path]
+
+    if "post_configs" in config_:
+        post_configs = config_["post_configs"]
+        for post_config in post_configs:
+            post_config = os.path.join(base_dir, post_config)
+            configs += [get_recursive_configs(post_config)]
+
+    return configs
+
+
```

Comment:
I would appreciate it if you could use `Path` instead of `os`
`os.path.dirname(path)` -> `path.parent`
`os.path.join(a, b)` -> a / b

Reply:
fixed